pkg/io/io.go

```diff
@@ -0,0 +1,67 @@
+package io
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"syscall"
+)
+
+var (
+	_ FileIO = (*MediaFileSystem)(nil)
+)
+
+// MediaFilesystem is the default implementation of file io using the os package
+type MediaFileSystem struct{}
+
+// Rename is a wrapper around os.Rename
+func (o *MediaFileSystem) Rename(source, target string) error {
+	return os.Rename(source, target)
+}
+
+// Open is a wrapper around os.Open
+func (o *MediaFileSystem) Open(name string) (*os.File, error) {
+	return os.Open(name)
+}
+
+// Create is a wrapper around os.Create
+func (o *MediaFileSystem) Create(name string) (io.WriteCloser, error) {
+	return os.Create(name)
+}
+
+// Create copies a file from a source path to a target path
+func (o *MediaFileSystem) Copy(source, target string) (int64, error) {
+	sourceFile, err := o.Open(source)
+	if err != nil {
+		return 0, err
+	}
+	defer sourceFile.Close()
+
+	targetFile, err := o.Create(target)
```

Comment:
What does this do if the target exists?

Reply:
looks like the default behavior is to overwrite the contents which is maybe not the behavior we want.. I think we should probably bail out of it exists. Thoughts?

Reply:
https://github.com/kasuboski/mediaz/pull/23/commits/2d5c0cc5f26b604e7068ae47d8eae6a675b90750